models/astrocyte_lr_1994.h

```diff
@@ -0,0 +1,498 @@
+  // ----------------------------------------------------------------
+
+  /**
+   * State variables of the model.
+   * @note Copy constructor required because of C-style array.
+   */
+  struct State_
+  {
+
+    /**
+     * Enumeration identifying elements in state array State_::y_.
+     * The state vector must be passed to GSL as a C array. This enum
+     * identifies the elements of the vector. It must be public to be
+     * accessible from the iteration function.
+     */
+    enum StateVecElems
+    {
+      IP3 = 0,
+      Ca,     // 1
+      h_IP3R, // 2
+      SIC,    // 3
+      DSIC,   // 4
+      STATE_VEC_SIZE
+    };
+
+    //! cell state, must be C-array for GSL solver
+    double y_[ STATE_VEC_SIZE ];
+
+    State_( const Parameters_& ); //!< Default initialization
+    State_( const State_& );
+
+    State_& operator=( const State_& );
+
+    void get( DictionaryDatum& ) const;
+    void set( const DictionaryDatum&, const Parameters_&, Node* );
+  };
+
+  // ----------------------------------------------------------------
+
+  /**
+   * Buffers of the model.
+   */
+  struct Buffers_
+  {
+    Buffers_( astrocyte_lr_1994& );                  //!< Sets buffer pointers to 0
+    Buffers_( const Buffers_&, astrocyte_lr_1994& ); //!< Sets buffer pointers to 0
+
+    //! Logger for all analog data
+    UniversalDataLogger< astrocyte_lr_1994 > logger_;
+
+    /** buffers and sums up incoming spikes/currents */
+    RingBuffer spike_exc_;
+    RingBuffer currents_;
+
+    /** GSL ODE stuff */
+    gsl_odeiv_step* s_;    //!< stepping function
+    gsl_odeiv_control* c_; //!< adaptive stepsize control function
+    gsl_odeiv_evolve* e_;  //!< evolution function
+    gsl_odeiv_system sys_; //!< struct describing system
+
+    // Since IntergrationStep_ is initialized with step_, and the resolution
+    // cannot change after nodes have been created, it is safe to place both
+    // here.
+    double step_;            //!< step size in ms
+    double IntegrationStep_; //!< current integration time step, updated by GSL
+
+    /**
+     * Input current injected by CurrentEvent.
+     * This variable is used to transport the current applied into the
+     * _dynamics function computing the derivative of the state vector.
+     * It must be a part of Buffers_, since it is initialized once before
+     * the first simulation, but not modified before later Simulate calls.
+     */
+    double I_stim_;
+
+    // values to be sent by SIC event
+    std::vector< double > sic_values;
+  };
+
+  // ----------------------------------------------------------------
+
+  // Access functions for UniversalDataLogger -------------------------------
+
+  //! Read out state vector elements, used by UniversalDataLogger
+  template < State_::StateVecElems elem >
+  double
+  get_y_elem_() const
+  {
+    return S_.y_[ elem ];
+  }
+
+  // ----------------------------------------------------------------
+
+  Parameters_ P_;
+  State_ S_;
+  Buffers_ B_;
```

Comment:
Add `Variables_ V_;` here with struct containing `max_Ca` and initialise in `pre_run_hook()`.

Reply:
It's done.